src/services/MetaService.ts

```diff
@@ -256,6 +263,12 @@ export class MetaService {
       const meta: any = this.memory[cleaned];
       if (!meta) {
         result.push(cleaned);
+      } else if (
+        meta.dataSpecialization === 'INLINE_EMBEDDED'
+        && meta.associatedEntity
+        && meta.associatedEntity.fields
+        && this.getSubFields(field).some(subField => !meta.associatedEntity.fields.find(aef => aef.name === subField))) {
+        result.push(cleaned);
```

Comment:
something i might have missed earlier is that we are pushing `cleaned` that seems wrong because all the default fields would have to be included, Maybe that is true for INLINE_EMBEDDED.

was think something more like 

``` javascript
  private missing(fields): string[] {
    if (!this.memory) {
      return fields;
    }
    if (fields && fields[0] === '*' && this.allFieldsLoaded) {
      return [];
    }
    const result: string[] = [];
    for (const field of fields) {
      const cleaned: string = this._clean(field);
      const meta: any = this.memory[cleaned];
      if (!meta) {
        // We can't push `cleaned` field because that wouldn't request subField
        result.push(field);
      } else if (meta.associatedEntity) {
        const aefs = meta.associatedEntity.fields.map(aef => aef.name);
        // filter out any existing associatedEntity fields
        // but this probably needs to be recursive or the subfields need to be cleaned
        const missingSubFields = this.getSubFields(field).filter(sub => !aefs.includes(sub));
        // construct field with missing sub fields
        result.push(`${cleaned}(${missingSubFields.join()})`);
      }
    }
    return result;
  }
```